### PR TITLE
Smoke Flamer

### PR DESCRIPTION
**Pull request by iloveloopers**
# About the pull request
adds a new flamer fuel tank that turns any reagent inside into chemsmoke when fired
fuel tank obtainable from army lathe (OT lathe)
only 1 reagent allowed inside the tank so its not overpowered
cannot be used with the nozzle
the smoke will always have a set ammount of units in it (somewhat high) so that it works effectively with the on_touch reactions

# Explain why it's good for the game
more flamer fuel variety
new ways to apply anti-xeno chems

# Testing Photographs and Procedure

https://www.youtube.com/watch?v=ePTXqXiK330

# Changelog
:cl:
add: Adds in a new incinerator smoke tank that turns any reagent inside it into chemsmoke when fired.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/projectiles/guns/flamer/flamer.dm [hunk 1]
```diff
@@ -229,6 +232,66 @@
...
 
+/obj/item/weapon/gun/flamer/proc/unleash_smoke(atom/target, mob/living/user)
+	last_fired = world.time
+	if(!current_mag || !current_mag.reagents || !current_mag.reagents.reagent_list.len)
+		return
+
+	var/source_turf = get_turf(user)
+	var/smoke_range = 7 // the max range the smoke will travel
+	var/distance = 0 // the distance traveled
+	var/use_multiplier = 2 // if you want to increase the ammount of units drained from the tank
```
Comment (Drulikar) [inline code/modules/projectiles/guns/flamer/flamer.dm:243]:
```suggestion
	var/smoke_range = 5 // the max range the smoke will travel
	var/distance = 0 // the distance traveled
	var/use_multiplier = 3 // if you want to increase the ammount of units drained from the tank
```

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -3225,6 +3225,10 @@ Defined in conflicts.dm of the #defines folder.
 		to_chat(user, SPAN_WARNING("This chemical will clog the nozzle!"))
 		return
 
+	if(istype(gun.current_mag, /obj/item/ammo_magazine/flamer_tank/smoke)) // you can't fire smoke like a projectile!
+		to_chat(user, SPAN_WARNING("\The [src] can't be used with this fuel tank!"))
```
Comment (Drulikar) [inline code/modules/projectiles/gun_attachables.dm:3229]:
```suggestion
		to_chat(user, SPAN_WARNING("[src] can't be used with this fuel tank!"))
```

File: code/modules/projectiles/magazines/flamer.dm [hunk 3]
```diff
@@ -90,7 +90,8 @@
 		to_chat(user, SPAN_WARNING("You can't mix fuel mixtures!"))
 		return
 
-	if(!to_add.intensityfire && to_add.id != "stablefoam")
+	var/is_smoke_tank = istype(src, /obj/item/ammo_magazine/flamer_tank/smoke)
+	if(!to_add.intensityfire && to_add.id != "stablefoam" && !is_smoke_tank)
```
Comment (Drulikar) [inline code/modules/projectiles/magazines/flamer.dm:94]:
```suggestion
	if(!to_add.intensityfire && to_add.id != "stablefoam" && !istype(src, /obj/item/ammo_magazine/flamer_tank/smoke))
```
Its not as clean but via short-circuiting it reduces how much type checking is done is most scenarios.

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by Skatuk** [comment]
So you know the disrupting research property, can it be combined with it somehow?

**Comment by iloveloopers** [comment]
> So you know the disrupting research property, can it be combined with it somehow?

if you have a chemical with the disrupting property, you can fill the smoke tank with it and it will work

**Review by Drulikar — APPROVED** [review]
I've not been made aware of any issues with this PR nor has it received any player feedback, but should something come up feel free to open another PR.

**Pull request merged**